rp2040-hal/src/adc.rs

```diff
@@ -259,3 +302,234 @@ where
         Ok(self.read(chan).into())
     }
 }
+
+/// Used to configure & build an [`AdcFifo`]
+///
+/// See [`Adc::build_fifo`] for details, as well as the `adc_fifo_*` [examples](https://github.com/rp-rs/rp-hal/tree/main/rp2040-hal/examples).
+pub struct AdcFifoBuilder<'a, const SHIFTED: bool> {
+    adc: &'a mut Adc,
+}
+
+impl<'a, const SHIFTED: bool> AdcFifoBuilder<'a, SHIFTED> {
+    /// Manually set clock divider integral and fractional parts
+    ///
+    /// Default: 0, 0
+    pub fn clock_divider(self, int: u16, frac: u8) -> Self {
```

Comment:
You could add an explanation to those parameters.

Also, in the examples (eg. https://github.com/rp-rs/rp-hal/pull/626/files#diff-54b5bbce718eaba891383cd18b03aa2b4533ef7b314c7584ec31bf880712a18bR114) it is difficult to understand why one needs such an odd divisor value.

Reply:
Fair enough :)

Added a bit of explanation, and a few example values

Reply:
"A bit"? That doc comment is great!